packages/shared/src/common/date.ts

```diff
@@ -108,3 +109,66 @@ export function sortDate(
     ? buildDayjs(date1).diff(buildDayjs(date2))
     : buildDayjs(date2).diff(buildDayjs(date1));
 }
+
+export function throwIfInvalidHl7Timestamp(input: string): boolean {
+  if (!input || input.length !== 14) {
+    throw new MetriportError("Invalid HL7 date string format: expected YYYYMMDDHHMMSS");
```

Comment:
I don't think we should throw `MetriportError` on validation, we want it eventually to be converted to a 4xx and not a 5xx. We've been using `BadRequestError` for that.

Reply:
addressed